### PR TITLE
Boilerplate doc

### PR DESCRIPTION
**Pull request by Ria**
Add documentation for boiler plate.

Can you please check if you find this sufficient?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,31 @@ Stellar uses serverless.com framework to deploy serverless functions to cloud.
 7. Benchmarking is performed.
 8. Once all the experiments are finished, the service is removed using the serverless.com framework. (`serverless remove`)
 
+### Deployment boilerplate outline
+Below are listed key packages and components that are most essential to serverless.com framework deployment.
+ - src/setup/
+   - extract-configuration.go
+     - Takes the experiment.json file and assigns according values to Go Configuration struct.
```
Comment (kk-min) [inline README.md:109]:
```suggestion
     - Reads the experiment.json file and assigns values accordingly to Go Configuration struct.
```

File: README.md [hunk 2]
```diff
@@ -102,6 +102,31 @@ Stellar uses serverless.com framework to deploy serverless functions to cloud.
...
+   - assign-endpoints.go
+     - Assigns endpoints and routes to each function deployed. Endpoints and routes are also stored in the Configuration struct. The enpoint values are extracted form the serverless framework deploy message
```
Comment (kk-min) [inline README.md:111]:
```suggestion
     - Assigns endpoints and routes to each function deployed. Endpoint values are extracted from the Serverless framework deploy message and stored in the Configuration struct.
```

File: README.md [hunk 3]
```diff
@@ -102,6 +102,31 @@ Stellar uses serverless.com framework to deploy serverless functions to cloud.
...
+   - serverless-config.go
+     - Creates the serverless.yml file based on the data in the experiment json file.
+
+ - src/setup/building/
+   - This package takes care of building Go binaries and Gradle projects for Java deployment.
```
Comment (kk-min) [inline README.md:116]:
```suggestion
   - This package takes builds Go binaries and Gradle projects for Java deployment.
```

File: README.md [hunk 4]
```diff
@@ -102,6 +102,31 @@ Stellar uses serverless.com framework to deploy serverless functions to cloud.
...
+ - src/setup/code-generation/
+   - This package generates Hello World provider-specific function source-code used for deployment.
+ - src/setup/deployment/packaging/
+   - This package takes care of creating filler files, zipping of the function package and creation of artifacts.
+ - src/setup/deployment/connection/
+   - This package contains legacy code from the original deployment implementation, which used provider specific APIs. Most of this package will be eventually obsolete, but we may get use of some of its functionalities.
```
Comment (kk-min) [inline README.md:122]:
```suggestion
   - This package contains legacy code from the original deployment implementation, which used provider specific APIs. Most of this package will be eventually obsolete and deprecated, but we may get use of some of its functionalities.
```

File: README.md [hunk 5]
```diff
@@ -102,6 +102,31 @@ Stellar uses serverless.com framework to deploy serverless functions to cloud.
...
+     - Assigns endpoints and routes to each function deployed. Endpoints and routes are also stored in the Configuration struct. The enpoint values are extracted form the serverless framework deploy message
+   - serverless-config.go
+     - Creates the serverless.yml file based on the data in the experiment json file.
+
+ - src/setup/building/
+   - This package takes care of building Go binaries and Gradle projects for Java deployment.
+ - src/setup/code-generation/
+   - This package generates Hello World provider-specific function source-code used for deployment.
+ - src/setup/deployment/packaging/
+   - This package takes care of creating filler files, zipping of the function package and creation of artifacts.
```
Comment (kk-min) [inline README.md:120]:
```suggestion
   - This package creates filler files, zips function packages and creates artifacts.
```

**Review by kk-min — APPROVED** [review]
Thanks! Left some suggestions

**Review by ypwong99 — APPROVED** [review]
Lgtm

**Review by ustiugov — CHANGES_REQUESTED** [review]
commit history is a mess, please make a single commit and sign it.

**Comment by ypwong99** [comment]
> commit history is a mess, please make a single commit and sign it.

squashed and signed on @Ria's behalf

**Pull request merged**